#!/usr/bin/env python
import roslib
PACKAGE = "param_server"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#format gen.add(name, type, level, description, default_value, min, max)
#min, max only apply for int/double types
group = Group()
gen.add_group("test")
gen.add("int_example", int_t, 0, "An example int parameter", 50,  0, 100)
gen.add("double_example", double_t, 0, "An example double parameter", 0.4,  0, 100)
gen.add("default_avoid_dist", double_t, 0, "brief description", 50,  0, 100)
gen.add("collision_avoid_velocity_scale", double_t, 0, "brief description", .5, 0,  1)
gen.add("string_example", str_t, 0,"An example string parameter", "Hello World")
gen.add("bool_example", bool_t, 0, "An example bool parameter", True)

exit(gen.generate(PACKAGE, "param_server", "Params"))
